Sol.java

```diff
@@ -0,0 +1,104 @@
+import java.util.*;
+
+public class Sol {
+    int[] in ;
+    HashMap < Integer, Integer > exp;
+    LinkedList < Move > moves_coll;
+    int zeropos;
+
+    public LinkedList<Move> move_cars(int[] in , int[] out) {
```

Comment:
I'd suggest using List<Move> instead of LinkedList<Move> (just an interface not the implementation)